crates/tools/js/benchmark/src/solidity-tests.ts

```diff
@@ -39,65 +74,202 @@ export async function setupForgeStdRepo() {
-    )
-    .map((a) => a.id);
+  const allResults = [];
+  const runs = new Map<string, number[]>();
+  const recordRun = recordTime.bind(null, runs);
 
-  const configs = {
+  for (const [name, samples] of Object.entries(TEST_SUITES)) {
+    for (let i = 0; i < samples; i++) {
+      let ids = testSuiteIds;
+      if (name !== TOTAL_NAME) {
+        ids = ids.filter((id) => id.name === name);
+      }
+
+      const start = performance.now();
+      const results = await runAllSolidityTests(artifacts, ids, config);
+      const elapsed = performance.now() - start;
+
+      const expectedResults = name === TOTAL_NAME ? TOTAL_EXPECTED_RESULTS : 1;
+      if (results.length !== expectedResults) {
+        throw new Error(
+          `Expected ${expectedResults} results for ${name}, got ${results.length}`
+        );
+      }
+
+      const failed = new Set();
+      for (const res of results) {
+        for (const r of res.testResults) {
+          if (r.status !== "Success" && !EXCLUDED_TESTS.has(r.name)) {
+            failed.add(
+              `${res.id.name} ${r.name} ${r.status} reason:\n${r.reason}`
+            );
+          }
+        }
+      }
+      if (failed.size !== 0) {
+        console.error(failed);
+        throw new Error(`Some tests failed`);
+      }
+
+      // Log to stderr so that it doesn't pollute stdout where we write the results
+      console.error(
+        `elapsed (s) on run ${i + 1}/${samples} for ${name}: ${displaySec(elapsed)}`
+      );
+
+      if (name === TOTAL_NAME) {
+        recordRun(TOTAL_NAME, elapsed);
+      } else {
+        if (results.length !== 1) {
+          throw new Error(
+            `Expected 1 result for ${name}, got ${results.length}`
+          );
+        }
+        recordRun(results[0].id.name, elapsed);
+      }
+
+      // Hold on to all results to prevent GC from interfering with the benchmark
+      allResults.push(results);
+    }
+  }
+
+  const measurements = getMeasurements(runs);
+
+  // Log info to stderr so that it doesn't pollute stdout where we write the results
+  console.error("median total elapsed (s)", displaySec(measurements[0].value));
+
+  console.log(JSON.stringify(measurements));
+}
+
+function getConfig(forgeStdRepoPath: string): SolidityTestRunnerConfigArgs {
+  return {
     projectRoot: forgeStdRepoPath,
+    rpcCachePath: RPC_CACHE_PATH,
+    fsPermissions: [
+      { path: forgeStdRepoPath, access: FsAccessPermission.ReadWrite },
+    ],
+    testFail: true,
+    rpcEndpoints: {
+      // These are hardcoded in the `forge-std` foundry.toml
+      mainnet:
+        "https://eth-mainnet.alchemyapi.io/v2/WV407BEiBmjNJfKo9Uo_55u0z0ITyCOX",
+      optimism_sepolia: "https://sepolia.optimism.io/",
+      arbitrum_one_sepolia: "https://sepolia-rollup.arbitrum.io/rpc/",
+    },
     fuzz: {
-      failurePersistDir: path.join(forgeStdRepoPath, "failures"),
+      seed: "0x1234567890123456789012345678901234567890",
```

Comment:
Maybe add a comment here to indicate that we are using a seed to ensure deterministic execution.

Reply:
Thanks, fixed in [08dacd2](https://github.com/NomicFoundation/edr/pull/686/commits/08dacd20152819aa9b765ffca7af96865aff498f)